src/FINDOLOGIC/Api/Definitions/Definition.php

```diff
@@ -0,0 +1,15 @@
+<?php
+
+namespace FINDOLOGIC\Api\Definitions;
+
+use ReflectionClass;
+
+abstract class Definition
+{
+    public static function getConstants()
```

Comment:
Automatically calculating the constants is convenient. However, reflection should only be used with great care, since you lose all kinds of static code analysis checks, IDE support etc. 
Also, the way this is implemented requires you to always keep in mind that you must never add any non-autocomplete-block constants to the class `BlockType`, otherwise they will appear in unexpected places.

I would prefer an explicit list of blocks like you had before (even though I may have originally suggested getting rid of it).

Reply:
But having a parent class with a common `getConstants` method certainly makes sense.

Reply:
Yeah but I do not think this is an issue. It makes adding new constants a lot easier to maintain.